### PR TITLE
Integrate QR code scanner

### PR DESCRIPTION
**Pull request by cketti**
image

Depends on #8208

### CODE REVIEW

**Comment by wmontwe** [comment]
I'm looking for resources to generate and test QR codes.

**Comment by cketti** [comment]
You can use the following payload for testing:

`[1,[1,2],[0,"imap.domain.example",993,3,2,"username"],[[[0,"smtp.domain.example",587,2,1,"username"],["user@domain.example","Firstname Lastname"]]]]`

Any QR code generator should do, e.g. https://danielgjackson.github.io/qrcodejs/

File: feature/migration/qrcode/build.gradle.kts [hunk 1]
```diff
@@ -10,6 +9,7 @@ android {
 
 dependencies {
     implementation(projects.core.common)
+    implementation(projects.legacy.ui.base)
 
     implementation(projects.core.ui.compose.designsystem)
     debugImplementation(projects.core.ui.compose.theme2.k9mail)
```
Comment (wmontwe) [inline feature/migration/qrcode/build.gradle.kts:15]: This should not be necessary as it's part of the debug api of the `designsystem` module.

**Comment by wmontwe** [comment]
The badging needs to be updated to due the new camera permission.

File: feature/migration/qrcode/src/main/kotlin/app/k9mail/feature/migration/qrcode/domain/usecase/QrCodeSettingsWriter.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+internal class QrCodeSettingsWriter(
+    private val context: Context,
+    private val xmlSettingWriter: XmlSettingWriter,
+) : UseCase.QrCodeSettingsWriter {
+    override fun write(accounts: List<Account>): Uri {
+        val file = getSettingsFile()
+        writeSettingsToFile(file, accounts)
+
+        val authority = "${context.packageName}.qrcode.settings"
+        return FileProvider.getUriForFile(context, authority, file)
```
Comment (wmontwe) [inline feature/migration/qrcode/src/main/kotlin/app/k9mail/feature/migration/qrcode/domain/usecase/QrCodeSettingsWriter.kt:20]: The file is created and overwritten correctly when importing new data. Nevertheless, it persists on the device until the app is removed or a new QR code is successfully scanned. I believe it should be automatically deleted after a successful import.

**Review by wmontwe — APPROVED** [review]
I think for the beta it works good enough, but it generates a lot of error logs during showing the QR scanning screen.

My phone started buzzing when placed on the table, seems that the auto focus goes wild. But that could be an issue of the camera integration.

**Review by wmontwe — APPROVED** [review]

**Pull request merged**